### PR TITLE
Bump ratatui to v0.26.1

### LINKED ISSUE #60: Update to ratatui 0.26
Hi, tui-textarea currently targets ratatui 0.24.0, but the latest version is 0.26.1. Could you update the dependency and publish a new release?

### PR DESCRIPTION
**Pull request by holly-hacker**
Resolves #60

### CODE REVIEW

File: Cargo.toml
Comment (rhysd) [inline Cargo.toml:45]: I'm not understanding why this change is necessary. `>=0.23.0` means version 0.23.0 or above hence it doesn't prevent updating ratatui to 0.26.1 in your local. Am I missing anything?

**Review by rhysd — CHANGES_REQUESTED** [review]

File: Cargo.toml
Comment (joshka) [reply Cargo.toml:45]: >= 0.23.0 means that the resolver will choose 0.26.2 (or the latest released version of Ratatui at any point when dependency resolution is triggered). This causes pain when an application specifies any earlier version of ratatui (e.g. 0.25.0). While tui-textarea does indeed actually support any version of ratatui up to that point, it would be better to instead specify the caret form: "0.26.2" in the dependencies here (or 0.26.0 as that will have the same effect and not prevent apps upgrading to a later compatible release).

```suggestion
ratatui = { version = "0.26.2", default-features = false, optional = true }
```

This pain point is generally discovered by application users attempting to install an application without the `--locked` flag, and several instances can be found in repos that use this crate. (I can link to these if you need examples).

This problem is described in more detail in the cargo book.

- https://doc.rust-lang.org/cargo/reference/resolver.html#version-incompatibility-hazards
- https://doc.rust-lang.org/cargo/reference/resolver.html#recommendations
- https://doc.rust-lang.org/cargo/reference/resolver.html#unexpected-dependency-duplication

The obvious downside to this is that it requires tui-textarea to be reasonably responsive to Ratatui major version bumps (e.g. 0.27.0) in order for apps to be able to upgrade. For this I'd suggest enabling dependabot and adding 1-2 extra maintainers to help triage / apply PRs when you're not available. (https://github.com/rhysd/tui-textarea/issues/65)

Comment (holly-hacker) [reply Cargo.toml:45]: To preface, I'm no expert on dependency resolution, I may be wrong here.

Ratatui is still v0.x so every increase of `x` in the version number may contain breaking changes (see the changes in this PR). Because of this, Cargo would include multiple versions of ratatui in your project which means that `tui-textarea` uses v0.23 types while your crate uses ratatui v0.26 types. Essentially, your build will break when you use any version above 0.23 because the types no longer match up. I think the usage of `>=` is only meant for ranges where there will be no breaking changes.

In the linked issue @joshka also listed the following items from Cargo's documentation that seem to discourage the use of `>=`:
- [Unexpected dependency duplication](https://doc.rust-lang.org/cargo/reference/resolver.html#unexpected-dependency-duplication)
- [Multiple version requirements](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#multiple-version-requirements)

Comment (holly-hacker) [reply Cargo.toml:45]: It seems that the `gitui` library is also blocked on this PR, implying they cannot upgrade ratatui with this version range: extrawurst/gitui#2101

Comment (joshka) [reply Cargo.toml:45]: > To preface, I'm no expert on dependency resolution, I may be wrong here.

Me neither, but I've seen this specific failure mode on enough Ratatui apps to spend a bit of time understanding why it causes failures. Choosing either 0.26.0 or 0.26.2 and not using `>=` is the most appropriate choice here for the benefit of developers of apps using this lib.

Comment (rhysd) [reply Cargo.toml:45]: Ah, I see. Thank you both for your explanations. I misunderstood the specification. I don't have time to work on this immediately. But I'll try to look into this some point in this week.

**Comment by rhysd** [comment]
Closing in favor of #69.

**Pull request closed without merging**
